csi/src/nodeplugin_grpc.rs

```diff
@@ -0,0 +1,97 @@
+//! The mayastor node plugin gRPC service
+//! This provides access to functionality that needs to be executed on the same
+//! node as a Mayastor CSI node plugin, but it is not possible to do so within
+//! the CSI framework. This service must be deployed on all nodes the the
+//! Mayastor CSI node plugin is deployed.
+use crate::freezefs;
+use freezefs::{freeze_volume, unfreeze_volume, FreezeFsError};
+use mayastor_node_plugin::*;
+use tonic::{transport::Server, Code, Request, Response, Status};
+
+pub mod mayastor_node_plugin {
+    tonic::include_proto!("mayastornodeplugin");
+}
+
+#[derive(Debug, Default)]
+pub struct MayastorNodePluginSvc {}
+
+fn freeze_fs_err_to_status(err: FreezeFsError) -> Status {
+    let msg = format!("{:?}", err);
```

Comment:
It would be better to implement From trait here.

Reply:
Done. https://github.com/openebs/Mayastor/pull/462/commits/7c714899d911a2f659d5a7861968ba08b1aa354b